build.sh

```diff
@@ -0,0 +1,149 @@
+            echo "                              (This is automatic, but can be done manually if needed.)"
+	        echo "    -jN                    ! Executes the 'make' command using N threads. Default is 1."
+            echo "    -h, --help             | Displays this help."
+            echo ""
+            exit 1
+            ;;
+
+        -r|--run)
+            RUN=1
+            shift
+            ;;
+
+        -d|--debug)
+            DEBUG=1
+            shift
+            ;;
+
+        -v|--valgrind)
+            VALGRIND=1
+            shift
+            ;;
+        
+        -g|--regenerate-cmake)
+            REGENERATE_CMAKE=1
+            shift
+            ;;
+
+        -j*)
+            MAKE_THREAD_COUNT=$arg
+            shift
+            ;;
+
+        *)
+            POSITIONAL+=("$1")
+            shift
+            ;;
+    esac
+done
+
+set -- "${POSITIONAL[@]}"
+
+# Enters into the build directory
+mkdir -p $BUILD_DIR
+cd $BUILD_DIR
+
+# Determines if the CMake files need to be regenerated to match the current OS
+if [[ -f "cmake-ostype" ]]; then
+    read -r CMAKE_OSTYPE < "cmake-ostype"
+    if [[ $CMAKE_OSTYPE != $OSTYPE ]]; then
+        REGENERATE_CMAKE=1
+    fi
+else
+    REGENERATE_CMAKE=1
+fi
+
+case $OSTYPE in
+    # MinGW / Windows...
+    "msys")
+        # Disables the valgrind argument.
+        if [[ $VALGRIND -eq 1 ]]; then
+            echo "Cannot valgrind in MinGW. Disabling this argument."
+            VALGRIND=0
+        fi
+
+        if [[ $REGENERATE_CMAKE -eq 1 ]]; then
+            echo "Regenerating CMake Files for MinGW/Windows..."
+            rm -rf ./**
+            cmake .. -G "MinGW Makefiles"
+            echo $OSTYPE > "cmake-ostype"
+        fi
+
+        echo "Building for MinGW/Windows..."
+        make $THREAD_MAKE_COUNT
+        
+        # Moves artifacts to release folder
+        mv ./_deps/sfml-build/lib/sfml-audio-2.dll $RELEASE_DIR
+        mv ./_deps/sfml-build/lib/sfml-graphics-2.dll $RELEASE_DIR
+        mv ./_deps/sfml-build/lib/sfml-network-2.dll $RELEASE_DIR
+        mv ./_deps/sfml-build/lib/sfml-system-2.dll $RELEASE_DIR
+        mv ./_deps/sfml-build/lib/sfml-window-2.dll $RELEASE_DIR
+        cp ./_deps/sfml-src/extlibs/bin/x64/openal32.dll $RELEASE_DIR
+        mv ./_deps/zlib-build/libzlib1.dll $RELEASE_DIR
+        mv ./OHWorkshopUploader.exe $RELEASE_DIR
+        mv ./SSVOpenHexagon.exe $RELEASE_DIR
+        ;;
+    
+    # Linux...
+    "linux-gnu")
+        if [[ $REGENERATE_CMAKE -eq 1 ]]; then
+            echo "Regenerating CMake Files for Linux..."
+            rm -rf ./*
+            cmake ..
+            echo $OSTYPE > "cmake-ostype"
+        fi
+
+        echo "Building for Linux..."
+        make $MAKE_THREAD_COUNT
+
+        mv SSVOpenHexagon $RELEASE_DIR/SSVOpenHexagon.exe
+        mv OHWorkshopUploader $RELEASE_DIR/OHWorkshopUploader.exe
+        ;;
+esac
```

Comment:
Maybe have a fallback case here for debugging that prints out if `$OSTYPE` is not anything we expected?

Reply:
Not a bad idea. And ideally we just build on this case statement when we want to support different OS's!